library/Booster/Pattern/ApplyEquations.hs

```diff
@@ -163,12 +181,19 @@ data EquationPreference = PreferFunctions | PreferSimplifications
     deriving stock (Eq, Show)
 
 runEquationM ::
+    Bool ->
     KoreDefinition ->
     Maybe LLVM.API ->
     EquationM a ->
-    Either EquationFailure (a, [EquationTrace])
-runEquationM definition llvmApi (EquationM m) =
-    fmap (fmap $ toList . trace) <$> runExcept $ runStateT m $ startState definition llvmApi
+    IO (Either EquationFailure (a, [EquationTrace]))
+runEquationM doTracing definition llvmApi (EquationM m) = do
+    endState <-
+        runStdoutLoggingT
```

Comment:
This call to `runStdoutLoggingT` doubles up the logging - the `respond` function in `Booster.JsonRpc` itself already runs in an `m` with  `MonadLoggerIO m` context.
The `runEquationM` type should be `MonadLoggerIO m => ... m (Either ...)`, otherwise we need to control the logging (filters etc) in two places to avoid discrepancies. One of which is that [we actually log to `stderr` in the server](https://github.com/runtimeverification/hs-backend-booster/blob/main/tools/booster/Server.hs#L94).